### PR TITLE
refactor: Address most Sonar complexity findings

### PR DESCRIPTION
**Pull request by bcollamore**

### CODE REVIEW

**Comment by bcollamore** [comment]
The diff is ugly, but the approach is (1) primarily, ExtractMethod, (2) in some cases, remove superfluous if clauses, and (3) in a couple cases replace with linq.

File: Philips.CodeAnalysis.Test/MsTest/TestContextAnalyzerTest.cs [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 using System;
+using System.Linq;
 using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
 using System.Text.RegularExpressions;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CodeFixes;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json.Serialization;
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Test/MsTest/TestContextAnalyzerTest.cs:12]: Where is Newtonsoft.Json used?

File: Philips.CodeAnalysis.Test/MsTest/TestContextAnalyzerTest.cs [hunk 2]
```diff
@@ -1,12 +1,15 @@
 ﻿// © 2022 Koninklijke Philips N.V. See License.md in the project root for license information.
 
 using System;
+using System.Linq;
 using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Test/MsTest/TestContextAnalyzerTest.cs:6]: Using not used

**Review by ynsehoornenborg — APPROVED** [review]

File: Philips.CodeAnalysis.Test/MsTest/TestContextAnalyzerTest.cs (hunk 1 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.Test/MsTest/TestContextAnalyzerTest.cs:12]: It's not.

File: Philips.CodeAnalysis.Test/MsTest/TestContextAnalyzerTest.cs (hunk 2 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.Test/MsTest/TestContextAnalyzerTest.cs:6]: Correct. I still need to wire up dotnet-format to the CI/CD pipeline. There's a lot of Code Cleanup stuff still lingering.

**Pull request merged**